src/header/common/access_control/allow_headers.rs

```diff
@@ -0,0 +1,25 @@
+use std::fmt::{mod};
+
+use header;
+use header::shared;
+
+#[deriving(Clone)]
+struct AccessControlAllowHeaders(pub Vec<String>);
+
+impl header::Header for AccessControlAllowHeaders {
+    #[inline]
+    fn header_name(_: Option<AccessControlAllowHeaders>) -> &'static str {
+        "Access-Control-Allow-Headers"
+    }
+
+    fn parse_header(raw: &[Vec<u8>]) -> Option<AccessControlAllowHeaders> {
+        shared::from_comma_delimited(raw).map(AccessControlAllowHeaders)
+    }
+}
+
+impl header::HeaderFormat for AccessControlAllowHeaders {
+    fn fmt_header(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        let AccessControlAllowHeaders(ref parts) = *self;
```

Comment:
Hyper has many variants to do the common task of "how to get the inner value of the header struct for formatting"

* `self[]` (used by `Accept-Encoding` and `Accept`, requires `deref!`)
* `let tm = **self;` (used by `Date`)
* `self.0` (used by `Authorization`)

I personally prefer the first form `self[]` but also `self.0` is good. We just should make sure we use the same everywhere in the code.

Try it out: http://is.gd/ZkmjYV